### PR TITLE
 [open-formulieren/open-forms#4222] Fix max files upload limit 

### PR DESCRIPTION
**Pull request by Viicos**
Partly fixes open-formulieren/open-forms#4222

I can't make the test to pass, it seems to not update the data when only calling `.upload` :/

### CODE REVIEW

File: src/formio/components/FileField.js [hunk 1]
```diff
@@ -184,12 +185,8 @@ class FileField extends Formio.Components.components.file {
 
   upload(files) {
     if (this.component.multiple && this.component.maxNumberOfFiles) {
-      // this.data.file contains files that may have already been uploaded, while the argument 'files' gives the
```
Comment (Viicos) [inline src/formio/components/FileField.js:187]: `data` is a mapping of component keys -> values, so `file` was only relevant if it was the actual key of the file component

File: src/formio/components/FileField.js [hunk 2]
```diff
@@ -184,12 +185,8 @@ class FileField extends Formio.Components.components.file {
...
-      // files that are being uploaded in this call
-      if (
-        files.length > this.component.maxNumberOfFiles ||
-        this.data?.file?.length >= this.component.maxNumberOfFiles
-      ) {
+      const existingFiles = get(this.data, this.component.key);
```
Comment (Viicos) [inline src/formio/components/FileField.js:188]: There might be a formio util function to do this without using the lodash implementation detail, but couldn't find one

Comment (sergei-maertens) [reply src/formio/components/FileField.js:188]: `this.dataValue` is a getter that looks up the value using lodash in the base `Component` class, so I'd use that.

**Review by sergei-maertens — CHANGES_REQUESTED** [review]
I don't know about the tests and which/how many promises are involved, but you may need to convert the tests to use `async`/`await` for more reliable results.

File: src/formio/components/FileField.js [hunk 3]
```diff
@@ -184,12 +184,7 @@ class FileField extends Formio.Components.components.file {
 
   upload(files) {
     if (this.component.multiple && this.component.maxNumberOfFiles) {
-      // this.data.file contains files that may have already been uploaded, while the argument 'files' gives the
-      // files that are being uploaded in this call
-      if (
-        files.length > this.component.maxNumberOfFiles ||
-        this.data?.file?.length >= this.component.maxNumberOfFiles
-      ) {
+      if (this.component.dataValue + files.length > this.component.maxNumberOfFiles) {
```
Comment (sergei-maertens) [inline src/formio/components/FileField.js:187]: `this.dataValue`, not `this.component.dataValue`

**Review by sergei-maertens — CHANGES_REQUESTED** [review]

File: src/jstests/formio/components/file.spec.js [hunk 4]
```diff
@@ -66,4 +66,30 @@ describe('Multiple File Component', () => {
...
+    Formio.createForm(element, formJSON)
+      .then(form => {
+        form.setPristine(false);
+        const component = form.getComponent('multipleFiles');
+        component.upload([{name: 'File 1', size: '50'}]);
+
+        component.upload([
+          {name: 'File 2', size: '50'},
+          {name: 'File 3', size: '50'},
+        ]);
```
Comment (sergei-maertens) [inline src/jstests/formio/components/file.spec.js:85]: internally this is called `redraw()`, which returns a promise but nowhere do they actually await this promise. So I'm expecting some race conditions here - what happens if you put some `sleep` calls before and after the `upload` calls?

Comment (Viicos) [reply src/jstests/formio/components/file.spec.js:85]: I can make it work by pushing an initial file in `component.dataValue`

**Review by sergei-maertens — APPROVED** [review]
I give up with Formio, this is good enough for me

**Pull request merged**